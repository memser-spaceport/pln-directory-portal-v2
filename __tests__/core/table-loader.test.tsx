import React from 'react';
import { render, screen } from '@testing-library/react';
import TableLoader from '../../components/core/table-loader';
import '@testing-library/jest-dom';
describe('TableLoader Component', () => {
  it('renders without crashing', () => {
    render(<TableLoader />);
    expect(screen.getByText('Loading more data...')).toBeInTheDocument();
  });

  it('renders the loader container with correct class', () => {
    render(<TableLoader />);
    expect(document.querySelector('.loader')).toBeInTheDocument();
  });

  it('renders the spinner with correct class', () => {
    render(<TableLoader />);
    expect(document.querySelector('.spinner')).toBeInTheDocument();
  });

  it('displays the correct loading text', () => {
    render(<TableLoader />);
    expect(screen.getByText('Loading more data...')).toBeVisible();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<TableLoader />);
    expect(asFragment()).toMatchSnapshot();
  });
});
