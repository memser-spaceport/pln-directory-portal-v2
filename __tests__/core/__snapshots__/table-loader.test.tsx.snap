// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableLoader Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-c6b1c38f6f66a557 loader"
  >
    <div
      class="jsx-c6b1c38f6f66a557 spinner"
    />
    <p
      class="jsx-c6b1c38f6f66a557"
    >
      Loading more data...
    </p>
  </div>
</DocumentFragment>
`;
